nibabel/tests/test_quaternions.py

```diff
@@ -55,129 +53,135 @@ def test_fillpos():
     w, x, y, z = nq.fillpositive(xyz)
-    yield assert_true, w == 1
+    assert w == 1
     # Or lists
     xyz = [0] * 3
     w, x, y, z = nq.fillpositive(xyz)
-    yield assert_true, w == 1
+    assert w == 1
     # Errors with wrong number of values
-    yield assert_raises, ValueError, nq.fillpositive, [0, 0]
-    yield assert_raises, ValueError, nq.fillpositive, [0] * 4
+    with pytest.raises(ValueError):
+        nq.fillpositive([0, 0])
+    with pytest.raises(ValueError):
+        nq.fillpositive([0] * 4)
     # Errors with negative w2
-    yield assert_raises, ValueError, nq.fillpositive, [1.0] * 3
+    with pytest.raises(ValueError):
+        nq.fillpositive([1.0] * 3)
     # Test corner case where w is near zero
     wxyz = nq.fillpositive([1, 0, 0])
-    yield assert_true, wxyz[0] == 0.0
+    assert wxyz[0] == 0.0
 
 
 def test_conjugate():
     # Takes sequence
     cq = nq.conjugate((1, 0, 0, 0))
     # Returns float type
-    yield assert_true, cq.dtype.kind == 'f'
+    assert cq.dtype.kind == 'f'
 
 
 def test_quat2mat():
     # also tested in roundtrip case below
     M = nq.quat2mat([1, 0, 0, 0])
-    yield assert_array_almost_equal, M, np.eye(3)
+    assert_array_almost_equal, M, np.eye(3)
     M = nq.quat2mat([3, 0, 0, 0])
-    yield assert_array_almost_equal, M, np.eye(3)
+    assert_array_almost_equal, M, np.eye(3)
     M = nq.quat2mat([0, 1, 0, 0])
-    yield assert_array_almost_equal, M, np.diag([1, -1, -1])
+    assert_array_almost_equal, M, np.diag([1, -1, -1])
     M = nq.quat2mat([0, 2, 0, 0])
-    yield assert_array_almost_equal, M, np.diag([1, -1, -1])
+    assert_array_almost_equal, M, np.diag([1, -1, -1])
     M = nq.quat2mat([0, 0, 0, 0])
-    yield assert_array_almost_equal, M, np.eye(3)
+    assert_array_almost_equal, M, np.eye(3)
 
 
-def test_inverse():
+def test_inverse_0():
     # Takes sequence
     iq = nq.inverse((1, 0, 0, 0))
     # Returns float type
-    yield assert_true, iq.dtype.kind == 'f'
-    for M, q in eg_pairs:
-        iq = nq.inverse(q)
-        iqM = nq.quat2mat(iq)
-        iM = np.linalg.inv(M)
-        yield assert_true, np.allclose(iM, iqM)
+    assert iq.dtype.kind == 'f'
+
+
+@pytest.mark.parametrize("M, q", eg_pairs)
+def test_inverse_1(M, q):
+    iq = nq.inverse(q)
+    iqM = nq.quat2mat(iq)
+    iM = np.linalg.inv(M)
+    assert np.allclose(iM, iqM)
 
 
 def test_eye():
     qi = nq.eye()
-    yield assert_true, qi.dtype.kind == 'f'
-    yield assert_true, np.all([1, 0, 0, 0] == qi)
-    yield assert_true, np.allclose(nq.quat2mat(qi), np.eye(3))
+    assert qi.dtype.kind == 'f'
+    assert np.all([1, 0, 0, 0] == qi)
+    assert np.allclose(nq.quat2mat(qi), np.eye(3))
 
 
 def test_norm():
     qi = nq.eye()
-    yield assert_true, nq.norm(qi) == 1
-    yield assert_true, nq.isunit(qi)
+    assert nq.norm(qi) == 1
+    assert nq.isunit(qi)
     qi[1] = 0.2
-    yield assert_true, not nq.isunit(qi)
+    assert not nq.isunit(qi)
 
 
-@slow
-def test_mult():
+@dec.slow
```

Comment:
Should see if the equivalent can be done with pure unittest/pytest constructs. My impression is that numpy didn't really intend a lot of its testing infrastructure to be a public API.